packages/bull-vault/src/BullStrategy.sol

```diff
@@ -17,83 +24,111 @@ contract BullStrategy is ERC20, LeverageBull, UniBull {
     using StrategyMath for uint256;
 
     uint256 private constant ONE = 1e18;
+    uint32 private constant TWAP = 420;
+
+    /// @dev set to true when redeemShortShutdown has been called
+    bool private hasRedeemedInShutdown;
+    /// @dev ETH:wSqueeth Uniswap pool
+    address private immutable ethWSqueethPool;
+    /// @dev ETH:USDC Uniswap pool
+    address private immutable ethUSDCPool;
+    /// @dev wPowerPerp address
+    address private immutable wPowerPerp;
 
     /// @dev Crab contract address
     address public immutable crab;
-    
-    /// @dev the cap in ETH for the strategy, above which deposits will be rejected
-    uint256 public strategyCap;
+    /// @dev PowerToken controller
+    address public immutable powerTokenController;
 
     /// @dev true if Bull was initialized
     bool public isInitialized;
-
+    /// @dev the cap in ETH for the strategy, above which deposits will be rejected
+    uint256 public strategyCap;
```

Comment:
we might want to cap this in units of crab? could be eth as well and check the total leverage balance?

Reply:
Yes, will add cap check in a future PR